anyway/infographics_utils.py

```diff
@@ -943,25 +949,105 @@ class MotorcycleAccidentsVsAllAccidentsWidget(Widget):
     def __init__(self, request_params: RequestParams):
         super().__init__(request_params, WidgetId.motorcycle_accidents_vs_all_accidents)
         self.rank = 20
-        self.text = {"title": "תאונות אופנועים קשות וקטלניות בכביש 20 בהשוואה לכל הארץ"}
+        self.road_number = request_params.location_info["road1"]
+        self.text = {"title": f"תאונות אופנועים קשות וקטלניות בכביש {int(self.road_number)} בהשוואה לכל הארץ"}
 
     # noinspection PyMethodMayBeStatic
     def is_in_cache(self) -> bool:
         return False
 
     def generate_items(self) -> None:
         self.items = (
-            MotorcycleAccidentsVsAllAccidentsWidget.motorcycle_accidents_vs_all_accidents_mock_data()
+            MotorcycleAccidentsVsAllAccidentsWidget.motorcycle_accidents_vs_all_accidents(
+                self.request_params.start_time, self.request_params.end_time, self.road_number)
         )
 
+    '''with ROAD_NUMBER as (
+        select 20 as val
+    )
+    SELECT 
+    LOCATION, vehicle, num_of_accidents/sum(num_of_accidents) 
+        over(PARTITION BY LOCATION):: decimal AS percentage 
+    FROM 
+    (SELECT CASE WHEN road1=(select * from ROAD_NUMBER) or road2 = (select * from ROAD_NUMBER) 
+                    THEN CONCAT('כביש ',(select * from ROAD_NUMBER)::text)  
+                 ELSE 'כל הארץ' END AS LOCATION, 
+    -- grouping of vehicles 
+    CASE WHEN involve_vehicle_type IN (8, 10, 19, 9) THEN 'אופנוע' ELSE 'אחר' END AS vehicle,
+    COUNT(DISTINCT provider_and_id) num_of_accidents 
+    FROM involved_markers_hebrew 
+    WHERE road_type in (3,4) -- only inter-city 
+    AND accident_severity in (1,2) GROUP BY 1,2 ORDER BY 3 DESC) as ACCIDENTS_COUNT'''
+
     @staticmethod
-    def motorcycle_accidents_vs_all_accidents_mock_data():  # Temporary for Frontend
+    def motorcycle_accidents_vs_all_accidents(start_time, end_time, road_number):
+        location_label = "location"
+        location_all = "כל הארץ"
+        location_road = f'כביש {int(road_number)}'
+        case_location = case([
+            ((InvolvedMarkerView.road1 == road_number) | (InvolvedMarkerView.road2 == road_number),
+             location_road)
+        ],
+            else_=literal_column(f"'{location_all}'")
+        ).label(location_label)
+
+        vehicle_label = "vehicle"
+        vehicle_other = "אחר"
+        vehicle_motorcycle = "אופנוע"
+        case_vehicle = case([(
+            InvolvedMarkerView.involve_vehicle_type.in_([8, 10, 19, 9]), literal_column(f"'{vehicle_motorcycle}'")
+        )],
+            else_=literal_column(f"'{vehicle_other}'")
+        ).label(vehicle_label)
+
+        query = get_query(table_obj=InvolvedMarkerView, filters={}, start_time=start_time,
+                          end_time=end_time)
+
+        query_columns = [InvolvedMarkerView.accident_id, InvolvedMarkerView.provider_and_id,
```

Comment:
InvolvedMarkerView.provider_code is used in order by (original query stated it as 3)

Reply:
The number 3 in the `ORDER BY 3 DESC` is used to point the newly created `percentage` column.

You can read more about it in:
[ORDER BY the numbers](https://sqlstudies.com/2013/06/17/order-by-the-numbers/)

Reply:
I'm quite sure it refers to the internal query, not the external one (it's within the parentheses). See here as it will highlight the parenthses:
https://app.redash.io/hasadna/queries/511391/source

Reply:
Yes, you are right -
Sorry for the incorrect comment.